model/src/main/kotlin/DependencyGraph.kt

```diff
@@ -41,25 +47,40 @@ import java.util.SortedSet
  * with [Scope] and [PackageReference] objects (with shared references, so keeping memory consumption low, too), so
  * that it can be processed in the usual ways.
  */
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
 data class DependencyGraph(
     /**
      * A list with the identifiers of the packages that appear in the dependency graph. This list is used to resolve
-     * the numeric indices contained in the [DependencyReference] objects.
+     * the numeric indices contained in the [DependencyGraphNode] objects.
      */
     val packages: List<Identifier>,
 
     /**
      * Stores the dependency graph as a list of root nodes for the direct dependencies referenced by scopes. Starting
      * with these nodes, the whole graph can be traversed. The nodes are constructed from the direct dependencies
-     * declared by scopes that cannot be reached via other paths in the dependency graph.
+     * declared by scopes that cannot be reached via other paths in the dependency graph. Note that this property
+     * exists for backwards compatibility only; it is replaced by the lists of nodes and edges.
      */
-    val scopeRoots: SortedSet<DependencyReference>,
+    val scopeRoots: SortedSet<DependencyReference> = sortedSetOf(),
```

Comment:
Using `@Deprecated` would probably be the best solution. Unfortunately, this breaks the build, because the field is still referenced in some places that now cause compiler warnings, and *detekt* is configured to treat warnings as errors.

Reply:
> _detekt_ is configured to treat warnings as errors.

Nit: It's not about _detekt_, but our [Kotlin compiler options](https://github.com/oss-review-toolkit/ort/blob/cb118d8e306310cec16877804a02a95ac3862613/build.gradle.kts#L225).

Reply:
Ok, then let's not add `@Deprecated`.